File: demo-vk-rest-scraping.java/src/main/java/demo/model/City.java
package demo.model;

import java.util.List;

public class City {

    private final int cid;
    private final String title;

    private Integer schoolsCount;
    private List<School> schools;

    private Integer universitiesCount;
    private List<University> universities;

    public City(Item item) {
        this.cid = item.getCid();
        this.title = item.getTitle();
    }

    public int getCid() {
        return cid;
    }

    public String getTitle() {
        return title;
    }

    public Integer getSchoolsCount() {
        return schoolsCount;
    }

    public List<School> getSchools() {
        return schools;
    }

    public void setSchools(List<School> schools) {
        this.schools = schools;
        this.schoolsCount = schools.size();
    }

    public Integer getUniversitiesCount() {
        return universitiesCount;
    }

    public List<University> getUniversities() {
        return universities;
    }

    public void setUniversities(List<University> universities) {
        this.universities = universities;
        this.universitiesCount = universities.size();
    }

    @Override
    public String toString() {
        return "City{" +
                "cid='" + cid + '\'' +
                ", title='" + title + '\'' +
                '}';
    }
}

File: demo-vk-rest-scraping.kotlin/src/main/kotlin/demo/model/City.kt
package demo.model

class City(item: Item) {

    val cid: Int
    val title: String

    var schoolsCount: Int? = null
        private set

    var schools: List<School>? = null
        set(schools) {
            field = schools
            this.schoolsCount = schools!!.size
        }

    var universitiesCount: Int? = null
        private set

    var universities: List<University>? = null
        set(universities) {
            field = universities
            this.universitiesCount = universities!!.size
        }

    init {
        this.cid = item.cid
        this.title = item.title!!
    }

    override fun toString(): String {
        return "City{" +
                "cid='" + cid + '\'' +
                ", title='" + title + '\'' +
                '}'
    }
}
